### PR TITLE
Extensions support

### PR DESCRIPTION
**Pull request by pzduniak**
Hello!

I'm willing to take https://github.com/graph-gophers/graphql-go/pull/293 to merge-quality. I will use this feature extensively in one of my projects so just let me know what I should exactly do (and where the changes should end up) and I'll do it in a somewhat timely manner.

FWIW I'm aware that the scalar extensions are missing, I'm not sure how complicated it'd be to get them to work. If this is just a matter of appending the decorators then I can get this in pretty quickly.

### CODE REVIEW

**Comment by abourget** [comment]
Do you have an example use of this feature?

**Comment by abourget** [comment]
Oh I see: https://github.com/graph-gophers/graphql-go/pull/345/files#diff-6c1c52311bdf50eb159f5ae695bbb5ecR175

File: internal/schema/schema.go [hunk 1]
```diff
@@ -159,6 +160,15 @@ type InputObject struct {
 	// TODO: Add a list of directives?
 }
 
+// Extension type defines a GraphQL type extension.
+// Schemas, Objects, Inputs and Scalars can be extended.
+//
+// https://facebook.github.io/graphql/draft/#sec-Type-System-Extensions
+type Extension struct {
+	Type NamedType
+	// TODO: Add a list og directives
```
Comment (pavelnikolov) [inline internal/schema/schema.go]: typo "list og..." -> "list of..."

Comment (pzduniak) [reply internal/schema/schema.go]: Oops, thought I corrected all typos!

**Comment by pzduniak** [comment]
Any thoughts on what significant part of the feature could be missing? I've been using my fork for a few weeks so far and I haven't found anything yet.

**Comment by pavelnikolov** [comment]
LGTM

File: internal/schema/schema_test.go [hunk 2]
```diff
@@ -165,6 +165,40 @@ func TestParse(t *testing.T) {
...
+				worldField := typ.Fields.Get("world")
+				if worldField == nil {
+					return fmt.Errorf("field %q not found", "world")
+				}
+				if worldField.Type.String() != "String!" {
+					return fmt.Errorf("field %q has an invalid type: %q", "world", worldField.Type.String())
+				}
+				return nil
+			},
+		},
```
Comment (pavelnikolov) [inline internal/schema/schema_test.go]: I want to see some more unit tests with more extension examples before merging this.

Comment (pavelnikolov) [reply internal/schema/schema_test.go]: I'll merge this and I'll add the missing unit tests myself. Thank you @pzduniak

**Comment by pzduniak** [comment]
Ha, I meant to write these tests today. Thanks a lot for merging this feature!

**Comment by pavelnikolov** [comment]
I already added the tests. Thank you for the contribution.

**Pull request merged**